### PR TITLE
Improve diagnostics for `KnownType`

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR adds diagnostic methods for `KnownTypes`.

### CODE REVIEW

File: base/src/main/java/io/spine/util/Predicates2.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Utilities for working with predicates.
+ */
+public class Predicates2 {
```
Comment (armiol) [inline base/src/main/java/io/spine/util/Predicates2.java:37]: Looks `final`.

File: base/src/main/java/io/spine/util/Predicates2.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    /**
+     * Prevents instantiation of this utility class.
+     */
+    private Predicates2() {
+    }
+
+    /**
+     * Retrieves a {@link Predicate} on a given type {@code T} which distinguish passed values
```
Comment (armiol) [inline base/src/main/java/io/spine/util/Predicates2.java:46]: "distinguishes"

File: base/src/main/java/io/spine/util/Predicates2.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     * by some property.
+     *
+     * <p>The predicate is satisfied (returns {@code true}) iff the result of applying the given
```
Comment (armiol) [inline base/src/main/java/io/spine/util/Predicates2.java:49]: "if"

File: base/src/main/java/io/spine/util/Predicates2.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+     * @param <K>
+     *         the type of the key
+     * @return a predicate on {@code T}
+     */
+    public static <T, K> Predicate<T> distinctBy(Function<T, K> selector) {
+        return new DistinctBy<>(selector);
+    }
+
+    /**
+     * The predicate which remember previously seen values and returns {@code true}
```
Comment (armiol) [inline base/src/main/java/io/spine/util/Predicates2.java:67]: "remembers"

**Review by armiol — APPROVED** [review]
LGTM with a couple of comments to address before merging.

File: base/src/main/java/io/spine/util/Predicates2.java (hunk 3 above)
Comment (alexander-yevsyukov) [reply base/src/main/java/io/spine/util/Predicates2.java:49]: “iff” means “if and only if”.

**Pull request merged**